### PR TITLE
Release 0.2.0

### PR DESCRIPTION
**Pull request by chuwy**

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,8 @@
+0.2.0 (not released yet)
+------------------------
+Web UI (#39)
+Embed Spray server (#53)
```
Comment (alexanderdean) [inline CHANGELOG]: There's many more tickets in this release no?

File: Vagrantfile [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Vagrant.configure("2") do |config|
 
   config.vm.box = "ubuntu/trusty64"
+  config.vm.network "public_network", bridge: 'en1: Wi-Fi (AirPort)'
```
Comment (alexanderdean) [inline Vagrantfile]: Remove this before release

File: project/BuildSettings.scala [hunk 3]
```diff
@@ -32,6 +31,20 @@ object BuildSettings {
...
+  // Settings specific for Schema Guru CLI
+  lazy val coreSettings = Seq[Setting[_]](
+    description           :=  "For deriving JSON Schemas from collections of JSON instances",
+
+    mainClass in (Compile, run) := Some("com.snowplowanalytics.schemaguru.Main")
+  )
+
+  // Settings specific for Schema Guru web UI
+  lazy val webuiSettings = Seq[Setting[_]](
+    description           :=  "Rich web user interface for Schema Guru",
```
Comment (alexanderdean) [inline project/BuildSettings.scala]: I'd change to "Web UI and server for Schema Guru"

File: CHANGELOG (hunk 1 above)
Comment (chuwy) [inline CHANGELOG]: Yes, much more.

File: src/main/scala/com.snowplowanalytics/schemaguru/Main.scala [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+import org.json4s.jackson.JsonMethods._
+
+// Argot
+import org.clapper.argot._
+import org.clapper.argot.ArgotConverters._
+
+// This library
+import utils._
+
+object Main extends App with FileSystemJsonGetters {
```
Comment (alexanderdean) [inline src/main/scala/com.snowplowanalytics/schemaguru/Main.scala]: http://stackoverflow.com/questions/24437423/in-scala-should-i-use-the-app-trait

File: src/main/scala/com.snowplowanalytics/schemaguru/generators/LevenshteinAnnotator.scala [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+import org.json4s._
+import org.json4s.JsonDSL._
+
+object LevenshteinAnnotator {
+  /**
+   * Alias type for set of string pairs
+   */
+  type KeyPairs = Set[Pair[String, String]]
+
+  // Strings shorter than thresholdLength do not comparing
```
Comment (alexanderdean) [inline src/main/scala/com.snowplowanalytics/schemaguru/generators/LevenshteinAnnotator.scala]: "do not comparing" -> "are not compared"

File: src/main/scala/com.snowplowanalytics/schemaguru/utils/JsonPathExtractor.scala [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+ */
+package com.snowplowanalytics.schemaguru.utils
+
+// scalaz
+import scalaz._
+import Scalaz._
+
+// jsonpath
+import com.fasterxml.jackson.databind.ObjectMapper
+import io.gatling.jsonpath.JsonPath
```
Comment (alexanderdean) [inline src/main/scala/com.snowplowanalytics/schemaguru/utils/JsonPathExtractor.scala]: Nice

File: src/main/scala/com.snowplowanalytics/schemaguru/utils/SelfDescribingSchema.scala [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.schemaguru.utils
+
+// json4s
+import org.json4s._
+import org.json4s.JsonDSL._
+
+case class SelfDescribingSchema(vendor: String, name: Option[String], version: Option[String]) {
```
Comment (alexanderdean) [inline src/main/scala/com.snowplowanalytics/schemaguru/utils/SelfDescribingSchema.scala]: Add a docstring to explain the case class

File: webui/src/main/resources/web/js/actions.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2012-2014 Snowplow Analytics Ltd. All rights reserved.
```
Comment (alexanderdean) [inline webui/src/main/resources/web/js/actions.js]: Date should just be 2014-2015, not 2012-

**Comment by alexanderdean** [comment]
This looks great! Only main feedback is that I'd like to see the webserver Scala code moved into its own sub-package codepath.

**Comment by chuwy** [comment]
Very good.
Now soon I'll push tests and we need to know what's wrong with 2.11 in Travis build.
Also I've noticed that I missed our special cases for duplicated keys (I'll fix it quickly).

File: src/main/scala/com.snowplowanalytics/schemaguru/Main.scala (hunk 4 above)
Comment (chuwy) [inline src/main/scala/com.snowplowanalytics/schemaguru/Main.scala]: I can't get it. What's wrong here?

Comment (alexanderdean) [inline src/main/scala/com.snowplowanalytics/schemaguru/Main.scala]: Actually - it's fine. You can ignore

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,18 @@
+# Dockerfile with running fat jar of Schema Guru Web UI
+
+# Ubuntu build with java 7 (https://index.docker.io/u/dockerfile/java/)
+FROM java:7
+
+MAINTAINER Anton Parkhomenko, mailbox@chuwy.me
```
Comment (alexanderdean) [inline Dockerfile]: Hey @chuwy, can you change maintainer to Snowplow Analytics, support@snowplowanalytics.com please?

Comment (chuwy) [inline Dockerfile]: Of course. Sorry, I didn't thought about that.

**Pull request merged**